### PR TITLE
[CI] Minor fixes to CI accompanying C++17 version change

### PR DESCRIPTION
**Pull request by NlightNFotis**
Currently the Rust API build for macOS when running on `develop` is failing, with the toolchain saying that it cannot find the file for `<algorithm>`. This points to a stale or otherwise compromised build configuration, so this is an experimental PR to change the environment (macOS) to a newer version (12 -> 13) in an attempt to ameliorate the situation.

### CODE REVIEW

File: .github/workflows/pull-request-check-rust-api.yaml [hunk 1]
```diff
@@ -62,14 +62,14 @@ jobs:
...
+          VERSION=$(cat src/config.inc | python3 -c "import sys,re;line = [line for line in sys.stdin if re.search(r'CBMC_VERSION = (\d+\.\d+\.\d+)', line)];sys.stdout.write(re.search(r'CBMC_VERSION = (\d+\.\d+\.\d+)', line[0]).group(1))")
           cd src/libcprover-rust;\
           cargo clean;\
           CBMC_INCLUDE_DIR=../../${{env.default_include_dir}} CBMC_LIB_DIR=../../${{env.default_build_dir}}/lib CBMC_VERSION=$VERSION cargo test -- --test-threads=1
 
 
-  check-macos-12-cmake-clang-rust:
-    runs-on: macos-12
+  check-macos-13-cmake-clang-rust:
+    runs-on: macos-13
```
Comment (tautschnig) [inline .github/workflows/pull-request-check-rust-api.yaml:72]: This change isn't actually necessary, is it?

File: .github/workflows/pull-request-check-rust-api.yaml [hunk 2]
```diff
@@ -101,8 +101,15 @@ jobs:
...
+          # (https://docs.rs/cc/latest/cc/struct.Build.html#method.cpp_link_stdlib)
+          CXXSTDLIB: c++
         run: |
-          VERSION=$(cat src/config.inc | python3 -c "import sys,re;line = [line for line in sys.stdin if re.search('CBMC_VERSION = (\d+\.\d+\.\d+)', line)];sys.stdout.write(re.search('CBMC_VERSION = (\d+\.\d+\.\d+)', line[0]).group(1))")
+          export MACOSX_DEPLOYMENT_TARGET=10.15
+          VERSION=$(cat src/config.inc | python3 -c "import sys,re;line = [line for line in sys.stdin if re.search(r'CBMC_VERSION = (\d+\.\d+\.\d+)', line)];sys.stdout.write(re.search(r'CBMC_VERSION = (\d+\.\d+\.\d+)', line[0]).group(1))")
           cd src/libcprover-rust;\
           cargo clean;\
-          CBMC_INCLUDE_DIR=../../${{env.default_include_dir}} CBMC_LIB_DIR=../../${{env.default_build_dir}}/lib CBMC_VERSION=$VERSION cargo test -- --test-threads=1
+          CBMC_INCLUDE_DIR=../../${{env.default_include_dir}} CBMC_LIB_DIR=../../${{env.default_build_dir}}/lib CBMC_VERSION=$VERSION cargo test --verbose -- --test-threads=1
```
Comment (tautschnig) [inline .github/workflows/pull-request-check-rust-api.yaml]: Probably no longer necessary?

File: .github/workflows/pull-request-check-rust-api.yaml [hunk 3]
```diff
@@ -101,8 +101,15 @@ jobs:
       # We won't be running any of the regular regression tests, as these are covered
       # by the other jobs already present in `pull-request-checks.yaml`.
       - name: Run Rust API tests
+        env:
+          # This is needed because an issue on the CI runner is preventing the link-cplusplus crate
+          # cxx.rs depends on to select the correct default (libc++, the clang++ standard library)
+          # on this platform.
+          # (https://docs.rs/cc/latest/cc/struct.Build.html#method.cpp_link_stdlib)
+          CXXSTDLIB: c++
```
Comment (tautschnig) [inline .github/workflows/pull-request-check-rust-api.yaml]: Is this change actually needed?

**Review by tautschnig — APPROVED** [review]

File: .github/workflows/pull-request-check-rust-api.yaml (hunk 1 above)
Comment (NlightNFotis) [reply .github/workflows/pull-request-check-rust-api.yaml:72]: It's not needed, but to be honest I'm inclined to keep it, just so we expand our testing targets a bit (On personal machines I'm on 14, as probably does @esteffin because of internal IT policy), and we're only testing 11 and 12 on the `pull-request-checks.yaml` file.

The only downside I can think of for not making this change is potential cache contention (as in, the size the cache requires, because I believe we're already quite close to our allowed limit, if not over it already). But I think even with this downside the upside is probably worth it.

Any reason *not* to keep this one in particular?

File: .github/workflows/pull-request-check-rust-api.yaml (hunk 2 above)
Comment (NlightNFotis) [reply .github/workflows/pull-request-check-rust-api.yaml]: You're right, this was just for debugging. Will go now.

File: .github/workflows/pull-request-check-rust-api.yaml (hunk 3 above)
Comment (NlightNFotis) [reply .github/workflows/pull-request-check-rust-api.yaml]: Not really - was a speculative fix but didn't work. Will drop.

**Review by esteffin — APPROVED** [review]

File: .github/workflows/pull-request-check-rust-api.yaml (hunk 1 above)
Comment (tautschnig) [reply .github/workflows/pull-request-check-rust-api.yaml:72]: No big deal, this PR is a mixed bag of CI changes already, so I think you are right that we should keep this.
